data-raw/england-aande.R

```diff
@@ -0,0 +1,91 @@
+# ---- Load libs ----
+library(tidyverse)
+library(readxl)
+library(devtools)
+library(httr2)
+library(lubridate)
+
+# ---- Load internal sysdata.rda file with URL's ----
+load_all(".")
+
+# ---- Function to download and clean ----
+scrape_data <- function(id, sheet, range, date, col_index) {
+  # get url from query_urls table
+  query_url <-
+    query_urls |>
+    filter(id == {{ id }}) |>
+    pull(query)
+
+  # download xls to as a temp file
+  download <- tempfile(fileext = ".xls")
+
+  request(query_url) |>
+    req_perform(download)
+
+  # read the xls file
+  raw <-
+    read_excel(
+      download,
+      sheet = sheet,
+      range = range
+    )
+
+  # clean the df
+  data <- raw |>
+    select(all_of(col_index)) |> # select relevant columns
+    rename(
+      code = 1, 
+      total_attendances = 2, 
+      attendances_over_4hours = 3, 
+      total_emergency_admissions = 4, 
+      emergency_admissions_over_4hours = 5
+    ) |> # rename columns
+    slice(-1) |> # remove first row
+    filter(!is.na(code)) |> # remove NAs
+    mutate_at(2:5, as.numeric) |> # change cols datatype to numeric
+    mutate(
+      pct_attendance_over_4hours = round(attendances_over_4hours / total_attendances, 2),
+      pct_emergency_admissions_over_4hours = round(emergency_admissions_over_4hours / total_emergency_admissions, 2),
+      date = date
+    ) # add percentage and date columns
+
+  return(data)
+}
+
+# ---- ICB level data ----
+# - Iterate over all data sets and return as a dataframe
+
+# Generate a dataframe with function arguments
+icb_df <-
+  tibble(
+    id = query_urls[which(query_urls$id == "nhs_accident_emergency_april_22"):(which(query_urls$id == "nhs_accident_emergency_april_22") + 12), ] |> pull(id),
+    sheet = rep("System Level Data", 13),
+    range = rep("B16:AA61", 13),
+    date = c("April 2022", "May 2022", "June 2022", "July 2022", "August 2022", "September 2022", "October 2022", "November 2022", "December 2022", "January 2023", "February 2023", "March 2023", "April 2023"),
+    col_index = rep(list(c(1, 6, 14, 24, 25)), 13)
+  )
+
+# Build dataframe with all months
+england_icb_accidents_emergency <- pmap_dfr(icb_df, scrape_data)
+
+# Save output to data/ folder
+usethis::use_data(england_icb_accidents_emergency, overwrite = TRUE)
```

Comment:
Something doesn't look right in the `pct_emergency_admissions_over_4hours` column. If you run the following code to examine the highest values in that column, you will notice that the first 42 results all have percentages way over 100%:

```r
england_icb_accidents_emergency |>
  select(code, pct_emergency_admissions_over_4hours) |> 
  arrange(desc(pct_emergency_admissions_over_4hours)) |> 
  print(n= 50)
```

I suspect that in some of the raw data, the percentages were not reported as decimal, and this has leaked into the calculations (but this is just a hypothesis!). Please take a look through the data/code and fix the error.

Reply:
A random rogue hidden cell in one of the files was the culprit!